uktrade/data-hub-frontend
# Add edit contact loop to send referral form

## Description of change

When users are adding a referral, this allows them to add a contact while editing the referral and to move back to their partially-completed form. This builds on a similar loop used in the Interaction form, and implements the referral form in Redux.

Some of the approach here is an interim solution until the add contact form is moved to React/Redux (storing form information in session storage; returning the new contact information via query string parameters)

## Test instructions

* Visit Data Hub and go to the Companies List.
* Select a company and click on 'Refer this company'
* When filling out the form, find the 'add a new contact' link in the 'Company contact (optional)' hint.
* Fill out the new contact form and click 'Add contact'
* Your new contact's name should appear in a green success flash message
* Continue filling out the the rest of the form


## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
